### PR TITLE
man-db: symlink commands without 'g' prefix into libexec/bin

### PR DESCRIPTION
**Pull request by lembacon**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Make `man-db` consistent with other formulae prefixed by `g`.

### CODE REVIEW

**Review by sjackman — APPROVED** [review]
Thanks, @lembacon!

**Comment by sjackman** [comment]
Ping @maxim-belkin

**Comment by maxim-belkin** [comment]
I guess I'm not sure why we need to do this. @lembacon, what is the goal of installing symlinks in `libexec`?

**Comment by sjackman** [comment]
So people can add `libexec/gnubin` to the front of their `PATH` to use `man` rather than `gman`.

**Review by sjackman — CHANGES_REQUESTED** [review]
Please add caveats.
 ```ruby
def caveats; <<~EOS
    Commands also provided by macOS have been installed with the prefix "g".
    If you need to use these commands with their normal names, you
    can add a "gnubin" directory to your PATH from your bashrc like:
      PATH="#{opt_libexec}/gnubin:$PATH"
  EOS
  end
```

**Comment by maxim-belkin** [comment]
> So people can add `libexec/gnubin` to the front of their `PATH` to use `man` rather than `gman`.

I'm confused. Why are we adding `g` prefix in the first place?

**Comment by sjackman** [comment]
The default name is `gman`. People can choose to instead use `man` to execute `gman` by either adding an alias or adding `libexec/gnubin` to their `PATH`.

**Comment by maxim-belkin** [comment]
> The default name is `gman`. People can choose to instead use `man` to execute `gman` by either adding an alias or adding `libexec/gnubin` to their `PATH`.

Small correction: the default name is `man` which we change to `gman` with the explanation that `man` is provided by macOS.

1. What I don't understand is why we first say "it is dangerous to provide `man` (and similar) executables because they're provided by macOS" and then say "OK, let's allow (advanced) users to do that". 

2. The trickery suggested in this PR is outside of the standard procedures for building `man-db` formula and is Homebrew- and macOS- specific. If there was a decision to follow this path, my comment would be the following: in the current state of Homebrew,  this is probably fine. But in the long run there should be  a method for doing that semi-automatically does that by simply providing names for binary files that have to be linked/copied/etc. If there was no such decision, I'd suggest rethink the strategy.

File: Formula/man-db.rb [hunk 1]
```diff
@@ -57,6 +58,25 @@ def install
     system "make", "CFLAGS=#{ENV.cflags}"
     system "make", "install"
 
+    # Symlink commands without 'g' prefix into libexec/gnubin and
+    # man pages into libexec/gnuman
+    %w[apropos catman lexgrog man mandb manpath whatis].each do |cmd|
+      (libexec/"gnubin").install_symlink bin/"g#{cmd}" => cmd
```
Comment (maxim-belkin) [inline Formula/man-db.rb]: `man-db` is not `gnu`. Having `gnu` prefix is deceptive.

File: Formula/man-db.rb [hunk 2]
```diff
@@ -57,6 +58,25 @@ def install
...
+    end
+    %w[accessdb].each do |cmd|
+      (libexec/"gnusbin").install_symlink sbin/"g#{cmd}" => cmd
```
Comment (maxim-belkin) [inline Formula/man-db.rb]: 1. This is a single file, so no need for 3 lines of code.
2. Same as above: using `gnu` is deceptive. How about `nongnu`?

**Comment by maxim-belkin** [comment]
I left couple of comments but they apply to other places in the PR as well -- I just didn't want to flood the PR with repetitive text.

**Comment by sjackman** [comment]
The approach taken by @lembacon here is the same as in the GNU tools.
See https://github.com/Homebrew/homebrew-core/blob/cb29320ca710ad2b358bd87e7d051659fce2f1e7/Formula/coreutils.rb#L81-L84
and also gawk.rb gnu-indent.rb gnu-sed.rb gnu-tar.rb gnu-time.rb gnu-units.rb gnu-which.rb grep.rb

I agree that `libexec/gnubin` and `libexec/gnuman` are a bit strange names, since the tool is not GNU. I suggest that the directories be named `libexec/bin` and `libexec/man`.

**Comment by sjackman** [comment]
I agree that `libexec/gnubin` and `libexec/gnuman` are a bit strange names, since the tool is not GNU. I suggest that the directories be named `libexec/bin` and `libexec/man`.

**Comment by ericbn** [comment]
I suggest there should be some guidelines about when to symlink binaries:
- in the path (`/usr/local/bin`, `/usr/local/sbin`) with the original name
- in the path, but adding a prefix to the original name, and then guidelines about the prefixes
- outside the path (`/usr/local/opt/`), and then guidelines about the path format
- when to mix more than one approach above in the same formula

I don't know what else is out there, but we already saw some back and forth in https://github.com/Homebrew/homebrew-core/pull/36494 and I believe that was partially because of the lack of a clear plan.

**Comment by sjackman** [comment]
Anything that doesn't shadow a macOS `/usr/bin/` executable is fair game for symlinking.
We've settled on the prefix `g` as the prefix, understanding that it's a totally arbitrary choice.
Outside the PATH you can call it whatever you want, so use the upstream names.

**Comment by lembacon** [comment]
@sjackman @maxim-belkin I've made some changes,

- Added caveats.
- Renamed `libexec/gnubin` and `libexec/gnuman` to `libexec/bin` and `libexec/man`.
- Codes simplified for directories with single file.

**Comment by CL-Jeremy** [comment]
Actually, since it's solely a user binary, what could be the advantage of symlinks over an alias in the user's profile? I find aliasing more intuitive and transparent than exploiting the libexec folder for this use,

**Comment by sjackman** [comment]
Thanks, @lembacon !

@CL-Jeremy Most users use only `man`, but there are in fact seven executables in `bin`:
`apropos catman lexgrog man mandb manpath whatis`

**Review by sjackman — APPROVED** [review]

**Comment by maxim-belkin** [comment]
Noticed that `findutils` has `bin`, `libexec/bin`, and `libexec/gnubin`

`bin/gupdatedb` sets `LC_ALL` to `C` and calls `libexec/bin/gupdatedb`
`libexec/gnubin/updatedb` is a symlink to `libexec/bin/gupdatedb`

😮

**Pull request closed without merging**